testhelper/terraform.go

```diff
@@ -16,11 +16,19 @@ import (
 
 // RemoveFromStateFile Attempts to remove resource from state file
 func RemoveFromStateFile(stateFile string, resourceAddress string) (string, error) {
+	return RemoveFromStateFileV2(stateFile, resourceAddress, "terraform")
+}
+
+// RemoveFromStateFileV2 Attempts to remove resource from state file
+// stateFile: The path to the state file
+// resourceAddress: The address of the resource to remove
+// tfBinary: The path to the terraform binary
+func RemoveFromStateFileV2(stateFile string, resourceAddress string, tfBinary string) (string, error) {
 	var errorMsg string
 	if files.PathContainsTerraformState(stateFile) {
 		stateDir := filepath.Dir(stateFile)
 		log.Printf("Removing %s from Statefile %s\n", resourceAddress, stateFile)
-		command := fmt.Sprintf("terraform state rm %s", resourceAddress)
+		command := fmt.Sprintf("%s state rm %s", tfBinary, resourceAddress)
```

Comment:
In the rare case that someone is not using our `TestOptionsDefault`, the binary might still be empty string which will fail on this part.

Terratest takes care of this for the other calls here: https://github.com/gruntwork-io/terratest/blob/master/modules/terraform/cmd.go#L46

If the binary is empty string it will not fail until it reaches this point (I think).